images/router/haproxy/conf/haproxy-config.template

```diff
@@ -84,16 +84,18 @@ frontend fe_sni
   bind 127.0.0.1:10444 ssl {{ if (len .DefaultCertificate) gt 0 }}crt {{.DefaultCertificate}}{{ else }}crt /var/lib/haproxy/conf/default_pub_keys.pem{{ end }} crt /var/lib/containers/router/certs accept-proxy
   mode http
 
-  # re-ssl?
-  acl reencrypt hdr(host),map(/var/lib/haproxy/conf/os_reencrypt.map) -m found
-  use_backend be_secure_%[hdr(host),map(/var/lib/haproxy/conf/os_tcp_be.map)] if reencrypt
+  # check re-encrypt backends first - from most specific to general path.
+  acl reencrypt base,map_beg(/var/lib/haproxy/conf/os_reencrypt.map) -m found
+
+  # Search from most specific to general path (host case).
+  use_backend be_secure_%[base,map_beg(/var/lib/haproxy/conf/os_reencrypt.map)] if reencrypt
```

Comment:
Yeah, it does mean a double lookup only in the case its in the re-encrypt map. 
Don't think ```if TRUE``` will work because the ```use_backend``` is only meant processed if the condition evaluates to true. If we always do it for ```if TRUE```, we get back to the other behavior of falling back to using the ```default_backend``` if its not found.